### PR TITLE
feat: to_char support for oracle and postgres

### LINKED ISSUE #623: Support TO_CHAR for Oracle
Currently, row hash validation with Oracle as the source or target creates some discrepancies with numeric data since casting to VARCHAR introduces some formatting issues. For example:

In Oracle:
`select cast(0.123 as varchar(4000)) from dual;` returns ".123"

In Postgres or BigQuery:
`select cast(0.123 as text);` returns "0.123"

We should introduce the TO_CHAR function to DVT so that users can apply that instead of a CAST to VARCHAR in the YAML config when running a row validation. This will allow for a successful validation:
`select to_char(0.123, 'FM90.099') from dual;`
"0.123"

We can use the SQLAlchemy function `sa.func.to_char(compiled_arg, format)` to accomplish this in a similar pattern to how we registered a hash function for Oracle [here](https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/third_party/ibis/ibis_addon/operations.py#L178).

### PR DESCRIPTION
**Pull request by sramsrinivasan**
Closes #623

### CODE REVIEW

**Comment by sramsrinivasan** [comment]
Hi @nehanene15  - I have created hr.test table with row_id and row_value columns in Oracle21 and public.test table with the same columns in guestbook database in PostgreSQL. Sample config file is as below - Can you please test my branch and then approve?

```
result_handler: {}
source: oracle21
target: postgres
validations:
- calculated_fields:
  - depth: 0
    field_alias: cast__row_id
    source_calculated_columns:
    - row_id
    target_calculated_columns:
    - row_id
    type: cast
  - depth: 0
    field_alias: to_char__row_value
    source_calculated_columns:
    - row_value
    target_calculated_columns:
    - row_value
    type: to_char
  - depth: 1
    field_alias: ifnull__cast__row_id
    source_calculated_columns:
    - cast__row_id
    target_calculated_columns:
    - cast__row_id
    type: ifnull
  - depth: 1
    field_alias: ifnull__to_char__row_value
    source_calculated_columns:
    - to_char__row_value
    target_calculated_columns:
    - to_char__row_value
    type: ifnull
  - depth: 2
    field_alias: rstrip__ifnull__cast__row_id
    source_calculated_columns:
    - ifnull__cast__row_id
    target_calculated_columns:
    - ifnull__cast__row_id
    type: rstrip
  - depth: 2
    field_alias: rstrip__ifnull__to_char__row_value
    source_calculated_columns:
    - ifnull__to_char__row_value
    target_calculated_columns:
    - ifnull__to_char__row_value
    type: rstrip
  - depth: 3
    field_alias: upper__rstrip__ifnull__cast__row_id
    source_calculated_columns:
    - rstrip__ifnull__cast__row_id
    target_calculated_columns:
    - rstrip__ifnull__cast__row_id
    type: upper
  - depth: 3
    field_alias: upper__rstrip__ifnull__to_char__row_value
    source_calculated_columns:
    - rstrip__ifnull__to_char__row_value
    target_calculated_columns:
    - rstrip__ifnull__to_char__row_value
    type: upper
  - depth: 4
    field_alias: concat__all
    source_calculated_columns:
    - upper__rstrip__ifnull__cast__row_id
    - upper__rstrip__ifnull__to_char__row_value
    target_calculated_columns:
    - upper__rstrip__ifnull__cast__row_id
    - upper__rstrip__ifnull__to_char__row_value
    type: concat
  - depth: 5
    field_alias: hash__all
    source_calculated_columns:
    - concat__all
    target_calculated_columns:
    - concat__all
    type: hash
  comparison_fields:
  - cast: null
    field_alias: hash__all
    source_column: hash__all
    target_column: hash__all
  filter_status: null
  filters: []
  format: table
  labels: []
  primary_keys:
  - cast: null
    field_alias: row_id
    source_column: row_id
    target_column: row_id
  random_row_batch_size: null
  schema_name: hr
  table_name: test
  target_schema_name: public
  target_table_name: test
  threshold: 0.0
  type: Row
  use_random_rows: false
```

File: data_validation/query_builder/query_builder.py [hunk 1]
```diff
@@ -306,6 +306,16 @@ def hash(config, fields):
                 how=how,
             )
 
+    @staticmethod
+    def to_char(config, fields):
+        fmt = "FM90.099"
```
Comment (nehanene15) [inline data_validation/query_builder/query_builder.py]: We should let the user specify the format and set "FM90.099" as the default similar to the ifnull() function in the query builder.

Something like this:
`fmt = config.get("to_char_fmt", "FM90.099")`

File: third_party/ibis/ibis_addon/operations.py [hunk 2]
```diff
@@ -188,13 +196,23 @@ def sa_format_hashbytes_postgres(translator, expr):
     hash_func = sa.func.sha256(convert)
     return sa.func.encode(hash_func, sa.sql.literal_column("'hex'"))
 
+def sa_format_to_char_oracle(translator, expr):
```
Comment (nehanene15) [inline third_party/ibis/ibis_addon/operations.py]: We can just make this one function `sa_format_to_char()` and set the OracleExprTranslator and PostgreSQLExprTranslator  registry to the same function.

**Review by nehanene15 — CHANGES_REQUESTED** [review]
Left a comment on allowing user to specify the format, and condensing the sa_format_to_char into one function.

File: third_party/ibis/ibis_addon/operations.py (hunk 2 above)
Comment (sramsrinivasan) [reply third_party/ibis/ibis_addon/operations.py]: Done

File: third_party/ibis/ibis_addon/operations.py [hunk 3]
```diff
@@ -66,6 +66,10 @@ class HashBytes(ValueOp):
     how = Arg(rlz.isin({"sha256", "farm_fingerprint"}))
     output_type = rlz.shape_like("arg", "binary")
 
+class to_char(ValueOp):
+    arg = Arg(rlz.one_of([rlz.value(dt.Decimal), rlz.value(dt.float64)]))
+    fmt = Arg(rlz.isin({"FM90.099"}))
```
Comment (sramsrinivasan) [inline third_party/ibis/ibis_addon/operations.py]: @nehanene15  - I am unable to validate for the fmt to be string. Tried many rules but nothing seem to work. So I kept it to rlz.isin as of now. Might need some help on this for more generic validation

**Review by nehanene15 — APPROVED** [review]

**Pull request closed without merging**